src/theory/bv/bitblast/lazy_bitblaster.cpp

```diff
@@ -57,38 +59,42 @@ uint64_t numNodes(TNode node, utils::NodeSet& seen)
 }
 }
 
+TLazyBitblaster::TLazyBitblaster(context::Context* c,
+                                 bv::TheoryBV* bv,
+                                 const std::string name,
+                                 bool emptyNotify)
+    : TBitblaster<Node>(),
+      d_bv(bv),
+      d_ctx(c),
+      d_nullRegistrar(new prop::NullRegistrar()),
+      d_nullContext(new context::Context()),
+      d_assertedAtoms(new (true) context::CDList<prop::SatLiteral>(c)),
+      d_explanations(new (true) ExplanationMap(c)),
+      d_variables(),
+      d_bbAtoms(),
+      d_abstraction(NULL),
+      d_emptyNotify(emptyNotify),
+      d_fullModelAssertionLevel(c, 0),
+      d_name(name),
+      d_statistics(name)
+{
+  d_satSolver.reset(
+      prop::SatSolverFactory::createMinisat(c, smtStatisticsRegistry(), name));
 
-TLazyBitblaster::TLazyBitblaster(context::Context* c, bv::TheoryBV* bv,
-                                 const std::string name, bool emptyNotify)
-  : TBitblaster<Node>()
-  , d_bv(bv)
-  , d_ctx(c)
-  , d_assertedAtoms(new(true) context::CDList<prop::SatLiteral>(c))
-  , d_explanations(new(true) ExplanationMap(c))
-  , d_variables()
-  , d_bbAtoms()
-  , d_abstraction(NULL)
-  , d_emptyNotify(emptyNotify)
-  , d_fullModelAssertionLevel(c, 0)
-  , d_name(name)
-  , d_statistics(name) {
-
-  d_satSolver = prop::SatSolverFactory::createMinisat(
-      c, smtStatisticsRegistry(), name);
-  d_nullRegistrar = new prop::NullRegistrar();
-  d_nullContext = new context::Context();
-  d_cnfStream = new prop::TseitinCnfStream(d_satSolver,
-                                           d_nullRegistrar,
-                                           d_nullContext,
-                                           options::proof(),
-                                           "LazyBitblaster");
+  d_cnfStream.reset(
+      new prop::TseitinCnfStream(d_satSolver.get(),
+                                 d_nullRegistrar.get(),
+                                 d_nullContext.get(),
+                                 options::proof(),
+                                 "LazyBitblaster"));
 
-  d_satSolverNotify = d_emptyNotify ?
-    (prop::BVSatSolverInterface::Notify*) new MinisatEmptyNotify() :
-    (prop::BVSatSolverInterface::Notify*) new MinisatNotify(d_cnfStream, bv,
-                                                            this);
+  d_satSolverNotify.reset(
+      d_emptyNotify
+          ? (prop::BVSatSolverInterface::Notify*)new MinisatEmptyNotify()
```

Comment:
Nit: I don't think that the conversion to `(prop::BVSatSolverInterface::Notify*)` is necessary.